### PR TITLE
add pykerberos, requests-kerberos, exchangelib

### PR DESCRIPTION
**Pull request by catern**
###### Motivation for this change

I want these packages

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [X] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [X] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/development/python-modules/exchangelib/default.nix [hunk 1]
```diff
@@ -0,0 +1,33 @@
+{ stdenv, fetchFromGitHub, buildPythonPackage,
+  lxml, tzlocal, python-dateutil, pygments, future, requests-kerberos,
+  defusedxml, cached-property, isodate, requests_ntlm, dnspython,
+  psutil, requests-mock, pyyaml
+}:
+
+buildPythonPackage rec {
+    pname = "exchangelib";
+    version = "1.11.4";
+    name = "${pname}-${version}";
```
Comment (FRidh) [inline pkgs/development/python-modules/exchangelib/default.nix]: The `name` attribute is added by `buildPython*` and should therefore be removed.

File: pkgs/development/python-modules/pykerberos/default.nix [hunk 2]
```diff
@@ -0,0 +1,20 @@
+{ stdenv, fetchPypi, buildPythonPackage, krb5 }:
+
+buildPythonPackage rec {
+    pname = "pykerberos";
+    version = "1.2.1";
+    name = "${pname}-${version}";
```
Comment (FRidh) [inline pkgs/development/python-modules/pykerberos/default.nix]: The `name` attribute is added by `buildPython*` and should therefore be removed.

File: pkgs/development/python-modules/requests-kerberos/default.nix [hunk 3]
```diff
@@ -0,0 +1,28 @@
+{ stdenv, fetchFromGitHub, buildPythonPackage, requests, pykerberos, mock }:
+
+buildPythonPackage rec {
+    pname = "requests-kerberos";
+    version = "0.12.0";
+    name = "${pname}-${version}";
```
Comment (FRidh) [inline pkgs/development/python-modules/requests-kerberos/default.nix]: The `name` attribute is added by `buildPython*` and should therefore be removed.

**Review by FRidh — CHANGES_REQUESTED** [review]

**Comment by FRidh** [comment]
@GrahamcOfBorg build pythonPackages.pykerberos pythonPackages.requests-pykerberos pythonPackages.exchangelib python3Packages.pykerberos python3Packages.requests-pykerberos python3Packages.exchangelib

**Comment by catern** [comment]
Ping on this? I made the changes you asked for

File: pkgs/development/python-modules/exchangelib/default.nix [hunk 4]
```diff
@@ -0,0 +1,32 @@
+{ stdenv, fetchFromGitHub, buildPythonPackage,
+  lxml, tzlocal, python-dateutil, pygments, future, requests-kerberos,
+  defusedxml, cached-property, isodate, requests_ntlm, dnspython,
+  psutil, requests-mock, pyyaml
+}:
+
+buildPythonPackage rec {
+    pname = "exchangelib";
```
Comment (FRidh) [inline pkgs/development/python-modules/exchangelib/default.nix]: 2 space indentation

File: pkgs/development/python-modules/pykerberos/default.nix [hunk 5]
```diff
@@ -0,0 +1,19 @@
+{ stdenv, fetchPypi, buildPythonPackage, krb5 }:
+
+buildPythonPackage rec {
+    pname = "pykerberos";
```
Comment (FRidh) [inline pkgs/development/python-modules/pykerberos/default.nix]: 2 space indentation

File: pkgs/development/python-modules/requests-kerberos/default.nix [hunk 6]
```diff
@@ -0,0 +1,27 @@
+{ stdenv, fetchFromGitHub, buildPythonPackage, requests, pykerberos, mock }:
+
+buildPythonPackage rec {
+    pname = "requests-kerberos";
```
Comment (FRidh) [inline pkgs/development/python-modules/requests-kerberos/default.nix]: 2 space indentation

**Review by FRidh — CHANGES_REQUESTED** [review]

**Comment by catern** [comment]
Ping? @FRidh

**Comment by FRidh** [comment]
```
Ran 0 tests in 0.000s
```
No tests are found. Either the test runner cannot find tests and needs to be patched, or there simply are no tests in which case the tests need to be disabled. Do include a comment explaining why the tests are disabled.

**Comment by catern** [comment]
There are no tests (besides one which tests against an existing Kerberos deployment), so I commented to that effect and set doCheck=false.

**Pull request merged**